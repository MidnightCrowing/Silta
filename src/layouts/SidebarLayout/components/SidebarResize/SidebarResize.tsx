import clsx from 'clsx'
import type { FC } from 'react'
import { memo } from 'react'

import type { SidebarResizeProps } from './SidebarResize.types'

export const SidebarResize: FC<SidebarResizeProps> = memo(({ isResizing, onMouseDown, ...props }) => (
  <div
    role="button"
    aria-label="resize panel"
    className={clsx(
      isResizing ? 'b-r-(3px $colorNeutralBackground5Pressed)' : '',
    )}
    z="3"
    onMouseDown={onMouseDown}
    {...props}
  />
))
