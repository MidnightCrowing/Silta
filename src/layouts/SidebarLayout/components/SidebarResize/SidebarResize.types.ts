import type { HTMLAttributes } from 'react'

export interface SidebarResizeProps extends HTMLAttributes<HTMLDivElement> {
  onMouseDown: () => void
  isResizing: boolean
}
