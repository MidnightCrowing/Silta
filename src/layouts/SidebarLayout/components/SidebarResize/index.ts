export { SidebarResize } from './SidebarResize'
export type { SidebarResizeProps } from './SidebarResize.types'
