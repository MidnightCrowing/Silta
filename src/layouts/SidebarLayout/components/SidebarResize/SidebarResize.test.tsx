import '@testing-library/jest-dom'

import { fireEvent, render } from '@testing-library/react'
import { describe, expect, it, vi } from 'vitest'

import { SidebarResize } from './SidebarResize'

describe('sidebarResize', () => {
  it('渲染默认状态', () => {
    const result = render(
      <SidebarResize
        onMouseDown={() => {}}
        isResizing={false}
      />,
    )
    expect(result.container).toMatchSnapshot()
  })

  it('渲染调整大小状态', () => {
    const result = render(
      <SidebarResize
        onMouseDown={() => {}}
        isResizing
      />,
    )
    expect(result.container).toMatchSnapshot()
  })

  it('点击时应该触发onMouseDown事件', () => {
    const onMouseDownMock = vi.fn()
    const { getByRole } = render(
      <SidebarResize
        onMouseDown={onMouseDownMock}
        isResizing={false}
      />,
    )
    const divElement = getByRole('button')
    fireEvent.mouseDown(divElement)
    expect(onMouseDownMock).toHaveBeenCalledTimes(1)
  })

  it('渲染额外的属性', () => {
    const { container } = render(
      <SidebarResize
        onMouseDown={() => {}}
        isResizing={false}
        absolute
      />,
    )
    const divElement = container.firstChild as HTMLElement
    expect(divElement).toHaveAttribute('absolute')
  })
})
