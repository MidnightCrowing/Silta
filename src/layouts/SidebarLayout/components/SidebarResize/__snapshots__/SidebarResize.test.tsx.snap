// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`sidebarResize > renders a default state 1`] = `
<div>
  <div
    aria-label="resize panel"
    class=""
    role="button"
    z="3"
  />
</div>
`;

exports[`sidebarResize > renders with resizing state 1`] = `
<div>
  <div
    aria-label="resize panel"
    class="b-r-3px b-r-$colorNeutralBackground5Pressed"
    role="button"
    z="3"
  />
</div>
`;

exports[`sidebarResize > 渲染调整大小状态 1`] = `
<div>
  <div
    aria-label="resize panel"
    class="b-r-3px b-r-$colorNeutralBackground5Pressed"
    role="button"
    z="3"
  />
</div>
`;

exports[`sidebarResize > 渲染默认状态 1`] = `
<div>
  <div
    aria-label="resize panel"
    class=""
    role="button"
    z="3"
  />
</div>
`;
